frontend/src/metabase/admin/performance/hooks/useInvalidateTarget.tsx

```diff
@@ -0,0 +1,47 @@
+import { useCallback } from "react";
+
+import { useDispatch } from "metabase/lib/redux";
+import { addUndo } from "metabase/redux/undo";
+import { CacheConfigApi } from "metabase/services";
+import type { CacheableModel } from "metabase-types/api";
+
+import { resolveSmoothly, isErrorWithMessage } from "../utils";
+
+export const useInvalidateTarget = (
+  targetId: number | null,
+  targetModel: CacheableModel,
+  { smooth = true, shouldThrowErrors = true } = {},
+) => {
+  const dispatch = useDispatch();
+  const invalidateTarget = useCallback(async () => {
+    if (targetId === null) {
+      return;
+    }
+    try {
+      const invalidate = CacheConfigApi.invalidate(
+        { include: "overrides", [targetModel]: targetId },
+        { hasBody: false },
+      );
```

Comment:
I understand this is being refactored from elsewhere, but ideally we should refactor this to RTKQuery in another PR.